mxcache-generation/src/test/java/com/maxifier/mxcache/template/impl/PInlineDependencyCache.template

```diff
@@ -68,6 +76,26 @@ public class #E#InlineDependencyCache extends #E#InlineCacheImpl implements Depe
             dependentNodes = new THashSet<Reference<DependencyNode>>();
         }
         dependentNodes.add(node.getSelfReference());
+        cleanupIfNeeded();
+    }
+
+    private void cleanupIfNeeded() {

```

Comment:
This class is a template, others are autogenerated from this template.

It doesn't seem reasonable to create some kind of helper for a single method or add one more class to the  hierarchy. 

Code generation is bad, I know. But there's no other way to have high performance with primitives in Java.